flocker/node/_docker.py

```diff
@@ -516,13 +516,17 @@ def add(self, unit_name, image_name, ports=None, environment=None,
         restart_policy_dict = self._serialize_restart_policy(restart_policy)
 
         def _create():
-            binds = {
-                volume.node_path.path: {
-                    'bind': volume.container_path.path,
-                    'ro': False,
-                }
+            binds = list(
+                # The "Z" mode tells Docker to "relabel file objects" on the
+                # volume.  This makes things work when SELinux is enabled, at
+                # least in the default configuration on CentOS 7.  See
+                # <https://docs.docker.com/reference/commandline/run/>, in the
+                # `--volumes-from` section (or just search for SELinux).
+                u"{}:{}:Z".format(
```

Comment:
Maybe this change is breaking Ubuntu? The test that times out is using a volume.